### PR TITLE
introduce docker-compose for sui devnet fullnode

### PR DESCRIPTION
**Pull request by tharbert**

### CODE REVIEW

**Comment by Clay-Mysten** [comment]
Should this go into `main` as well or first?

File: docker/fullnode/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+version: "3.9"
+services:
+  fullnode:
+    image: mysten/sui-node:devnet-0.4.0
+    ports:
```
Comment (allan-bailey) [inline docker/fullnode/docker-compose.yaml:6]: need to add here the 'expose' ports section for those 2 ports.

**Comment by bmwill** [comment]
Yeah @tharbert is there an equivalent pr for main? I’d actually probably just have the pr against main and not do anything against devnet since we should be doing another release soon

**Comment by bmwill** [comment]
@tharbert actually looks like it may be cleaner to just manually move your commit to be based on main given looking at this it seems to be pulling in some extra commits from the devnet branch that we don't want

File: docker/fullnode/docker-compose.yaml (hunk 1 above)
Comment (tharbert) [reply docker/fullnode/docker-compose.yaml:6]: done, thanks

**Comment by tharbert** [comment]
@bmwill Take another look now please, I think we are good.

**Review by allan-bailey — APPROVED** [review]

**Comment by Clay-Mysten** [comment]
Hi Tom, Allan, and Brandon, how should we recommend this in our fullnode doc?
https://github.com/MystenLabs/sui/blob/main/doc/src/build/fullnode.md

Can it be placed in parallel to building from source? Perhaps above it? What are the ramifications of using it instead of building it from source? Thanks!

**Pull request merged**